composer.json

```diff
@@ -6,27 +6,28 @@
   "description": "Encrypted symfony entity's by verified and standardized libraries",
   "require": {
     "php": "^7.2",
-    "paragonie/halite": "^4.3",
-    "paragonie/sodium_compat": "^1.5",
+    "paragonie/halite": "^4.6",
     "doctrine/orm": "^2.5",
     "symfony/property-access": "^4.1",
     "symfony/dependency-injection": "^4.1",
     "symfony/yaml": "^4.1",
     "symfony/http-kernel": "^4.1",
     "symfony/config": "^4.1"
   },
-  "license": "MIT",
   "require-dev": {
     "phpunit/phpunit": "^6.5",
-    "defuse/php-encryption": "^2.1"
+    "defuse/php-encryption": "^2.1",
+    "ext-sodium": "*"
```

Comment:
this means people without the sodium extension can't develop on the bundle anymore. i guess thats no big issue, just wanted to mention it

Reply:
Very good point. The reason for adding it is that the current tests suite failed without sodium extenstion installed:
```
1) Ambta\DoctrineEncryptBundle\Tests\Unit\Encryptors\HaliteEncryptorTest::testEncrypt
SodiumException: This is not implemented, as it is not possible to securely wipe memory from PHP
/home/travis/build/sh41/DoctrineEncryptBundle/vendor/paragonie/sodium_compat/src/Compat.php:2444
/home/travis/build/sh41/DoctrineEncryptBundle/vendor/paragonie/sodium_compat/lib/php72compat.php:849
/home/travis/build/sh41/DoctrineEncryptBundle/vendor/paragonie/halite/src/HiddenString.php:101
/home/travis/build/sh41/DoctrineEncryptBundle/src/Encryptors/HaliteEncryptor.php:32
/home/travis/build/sh41/DoctrineEncryptBundle/Tests/Unit/Encryptors/HaliteEncryptorTest.php:18
Caused by
SodiumException: This is not implemented, as it is not possible to securely wipe memory from PHP
```

I think the test is valid and I think that the exception is valuable information for users of the polyfill to understand that there is some functionality that they don't get with the polyfill. 

The alternative might be to remove the dependency from `dev` and explicitly test for the exception being thrown. Perhaps there should be tests with the extension and tests without it? Verifying the behaviour is as expected in each case.

Reply:
It looks like libsodium is a required installed for certain functions of the Halite library to work correctly: https://github.com/paragonie/halite/issues/99

but users of this bundle have the option not to use Halite, so you're right it shouldn't be required to develop on it. I've moved to suggested in composer to give people a clue that they may need it.  

I've also added conditional skips to the tests that rely upon sodium being installed so that they won't fail when it isn't enabled. I added a version of the travis tests that does not have libsodium and an additional test that checks the appropriate exceptions are thrown when libsodium isn't installed.